tensorflow_addons/utils/types.py

```diff
@@ -46,7 +47,12 @@
 Regularizer = Union[None, dict, str, Callable, tf.keras.regularizers.Regularizer]
 Constraint = Union[None, dict, str, Callable, tf.keras.constraints.Constraint]
 Activation = Union[None, str, Callable]
-Optimizer = Union[tf.keras.optimizers.Optimizer, str]
+if importlib.util.find_spec("tensorflow.keras.optimizers.legacy") is not None:
+    Optimizer = Union[
+        tf.keras.optimizers.Optimizer, tf.keras.optimizers.legacy.Optimizer, str
```

Comment:
Same as above

Reply:
This I am using the name "Optimizer" to keep it unchanged.